core/src/main/java/org/mule/runtime/core/internal/processor/strategy/BlockingProcessingStrategyFactory.java

```diff
@@ -81,5 +81,21 @@ public ReactiveProcessor onProcessor(ReactiveProcessor processor) {
       }
     }
 
+    /**
+     * This strategy adds a Mono.block call in order to preserve the thread, because it's a precondition to transactions to work.
```

Comment:
precondition *for* transactions to work. I would remove the comma before because and the split the two last sentences.
"It  allows us to make a performance optimization for those operations.  This method is intended to detect which operations..."